### DISCUSSION TITLE
Is it possible to push a docker-compose file along with the Dockerfiles as an image?

### DISCUSSION THREAD

**Discussion by BretFisher** [BretFisher/ama#175] [Q&A]
That's currently not possible. You'd need to keep your compose files along with your Dockerfile in a git repo.  An image is a *output* of building your app from the Dockerfile, and isn't meant to be a transport for dockerfile/compose files. Images do have metadata and `docker image history` can technically give you Dockerfile commands, but it's not really a replacement for them.

**Comment by clemenko** [comment]
Why not use a "LABEL" in the Dockerfile that points to the git repo with the Dockerfile and Compose file? We used this in past. Then you can `docker inspect` the image to retrieve the git address. 

`docker inspect clemenko/flask_demo | jq .[].ContainerConfig.Labels.GITREPO`

**Comment by clemenko** [comment]
Ha... autoopen the url on a mac.
`open $(docker inspect clemenko/flask_demo | jq .[].ContainerConfig.Labels.GITREPO -r)`

**Comment by BretFisher** [comment]
That's a good idea for documentation purposes @clemenko.  The unofficial standard for that label is `vcs-url` from http://label-schema.org which is trying to set best practices around common container labels.

**Comment by clemenko** [comment]
Cool. didn't know about that site.

**Comment by BretFisher** [comment]
Note this is now possible with Docker's new "App" tool: https://github.com/docker/app
